userspace/libsinsp/filter.h

```diff
@@ -213,6 +213,8 @@ class sinsp_filter_factory : public gen_event_filter_factory
 
 	gen_event_filter_check *new_filtercheck(const char *fldname);
 
+	std::list<gen_event_filter_factory::filter_fieldclass_info> get_fields() override;
+
```

Comment:
I assume this is the only subclass?

Reply:
In the libraries, yes. In falco there's another subclass for "json_events", which are used by the k8s audit rules: https://github.com/leogr/falco/blob/new/plugin-system-api-additions/userspace/engine/json_evt.h#L376.

And after these changes are all merged, we'll be creating a subclass for plugins, so a given plugin can export the set of valid fields used by that plugin.